src/drt/test/fixture.cpp

```diff
@@ -83,6 +83,31 @@ void Fixture::addLayer(frTechObject* tech,
   tech->addLayer(std::move(layer));
 }
 
+odb::dbInst* Fixture::createDummyInst()
+{
+  utl::Logger* logger = new utl::Logger();
+  odb::dbDatabase* db = odb::dbDatabase::create();
+  db->setLogger(logger);
+  odb::dbTech* tech = odb::dbTech::create(db, "tech");
+  odb::dbTechLayer::create(tech, "L1", dbTechLayerType::MASTERSLICE);
+  odb::dbLib* lib = odb::dbLib::create(db, "lib1", tech, ',');
+  odb::dbChip* chip = odb::dbChip::create(db);
+  odb::dbBlock::create(chip, "simple_block");
+  odb::dbMaster* master = odb::dbMaster::create(lib, "dummy");
+  master->setWidth(1000);
+  master->setHeight(1000);
+  master->setType(dbMasterType::CORE);
+  odb::dbMTerm::create(master, "a", dbIoType::INPUT, dbSigType::SIGNAL);
+  odb::dbMTerm::create(master, "b", dbIoType::INPUT, dbSigType::SIGNAL);
+  odb::dbMTerm::create(master, "c", dbIoType::OUTPUT, dbSigType::SIGNAL);
+  master->setFrozen();
+  auto ptr_db_inst = std::make_unique<odb::dbInst>();
+  odb::dbInst* db_inst = ptr_db_inst->create(chip->getBlock(), master, "dummy");
+  dbTransform trans = dbTransform();
+  db_inst->setTransform(trans);
+  return db_inst;
+}
```

Comment:
This looks weird. I see that in this function you're creating a database, library, chip, block, master and inst. They are all separate from anything related to the actual design just to set a correct transform. The correct approach would be to modify the infrastructure for the test suite altogether. The setup for the test suite is done at Fixture::makeDesign. in that function you could initialize a database, library, chip and a block. in Fixture::makeMacro, you could create the corresponding dbMaster. Same would apply in makeMacroPin. finally, in makeInst create the corresponding dbInst as well. 
In the end, you should have only one db, tech, chip, block, and library for the design. and they all should be meaningful.

Reply:
I'll be sure to make fix this. Most of the code was taken from `dbDatabase* createSimpleDB`.